pkg/rke2/rke2_linux.go

```diff
@@ -199,3 +206,22 @@ func hostnameFQDN() (string, error) {
 
 	return strings.TrimSpace(b.String()), nil
 }
+
+func hostnameFromMetadataEndpoint() string {
+	resp, err := http.Get("http://169.254.169.254/latest/metadata/local-hostname")
+	if err != nil {
+		return ""
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return ""
```

Comment:
nit: can we leave a warning or a debug log to just report the error if its there

Reply:
+1 on debug logging it.

Reply:
Updated.